packages/stack/proxy/src/proxy.js

```diff
@@ -18,41 +18,46 @@ export class Proxy {
     this.logger = options.logger;
     this.vms = options.vms;
     this.app = null;
-    this.requestManager;
+    this.endpoint = options.endpoint;
+    if (options.endpoint === constants.blockchain.vm) {
+      this.endpoint = this.vms[this.vms.length - 1]();
+    }
+    this.isWs = options.isWs;
   }
 
-  async serve(endpoint, localHost, localPort, ws) {
-    if (endpoint === constants.blockchain.vm) {
-      endpoint = this.vms[this.vms.length - 1]();
-    }
-    this.requestManager = new Web3RequestManager.Manager(endpoint);
+  async nodeReady() {
+    const requestManager = new Web3RequestManager.Manager(this.endpoint);
 
     try {
-      await this.requestManager.send({ method: 'eth_accounts' });
+      await requestManager.send({ method: 'eth_accounts' });
     } catch (e) {
-      throw new Error(__('Unable to connect to the blockchain endpoint'));
+      throw new Error(__(`Unable to connect to the blockchain endpoint on ${this.endpoint}`));
     }
 
+    // disconnect socket connection (NO-OP when HTTP)
+    requestManager.provider.disconnect();
```

Comment:
If the `send()` in the `try` block fails and we end up in the `catch` block where we re-throw an error, do we actually end up in line 38?

Reply:
Moved the `.disconnect()` in to the `try` block.